### DISCUSSION TITLE
Question: How to handle errors in `build` for Notifier?

### DISCUSSION THREAD

**Discussion by tadaspetra** [rrousselGit/riverpod#2517] [Q&A] [answered]
In the example code on the v2 docs, there is a `_fetchTodo()`. Let's say this function call failed, what would be the proper way to handle this in the `build` method since the build method doesn't return an `AsyncValue`

https://github.com/rrousselGit/riverpod/blob/a89320db5492522681b004039262cdfcbcac4dc6/website/docs/providers/notifier_provider/remote_todos/raw.dart#L61

**Comment by rrousselGit** [comment, marked as answer]
Do nothing. Let it throw.

The error will be converted into an AsyncError automatically.

**Comment by tadaspetra** [comment]
That makes sense. This is not necessarily a problem, but this situation would always be marked as an "Uncaught Exception" by the IDE, even though technically it is being handled behind the scenes. Am I correct in that statement?

**Reply by tadaspetra** [reply]
Moved this to a new discussion, as the original question was answered https://github.com/rrousselGit/riverpod/discussions/2523
